sphinx/writers/latex.py

```diff
@@ -1204,7 +1206,7 @@ def depart_table(self, node):
             endmacro = '\\end{tabulary}\n\n'
         if self.table.colspec:
             self.body.append(self.table.colspec)
-        elif self.table.colwidths:
+        elif self.table.colwidths and 'colwidths-given' in self.table.classes:
```

Comment:
I needed to change also at line https://github.com/sphinx-doc/sphinx/pull/3389/files#diff-81c44cde8d80608ae558b93878f9be8aR1199 to let test pass